planner/core/logical_plan_builder.go

```diff
@@ -2784,7 +2796,7 @@ func (b *PlanBuilder) buildUpdateLists(ctx context.Context, tableList []*ast.Tab
 		}
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.UpdatePriv, dbName, col.OrigTblName.L, "", nil)
 	}
-	return newList, p, nil
+	return newList, p, allAssignmentsAreConstant, nil
```

Comment:
When will `allAssignmentsAreConstant`  be true?
Forget  set  `allAssignmentsAreConstant` to  true first?